optuna/samplers/_tpe/probability_distributions.py

```diff
@@ -1,4 +1,5 @@
-from typing import List
+from __future__ import annotations
+
 from typing import NamedTuple
 from typing import Union
```

Comment:
Could we remove this line and change the code using `Union` as following? (I confirmed that this change works in local python3.11 environment...)
```
_BatchedDistributions = (
    _BatchedCategoricalDistributions
    | _BatchedTruncNormDistributions
    | _BatchedDiscreteTruncNormDistributions
)
```

Reply:
I can check whether the check passes with this modification, but if I remember correctly, it does not work.

Reply:
It did not work, so I will revert the change!
https://github.com/optuna/optuna/actions/runs/9059459819/job/24887097949?pr=5443

Reply:
Thank you for trying it!